server/proxy/channels/pf_channel_drdynvc.c

```diff
@@ -93,16 +107,27 @@ static pServerDynamicChannelContext* DynamicChannelContext_new(pServerContext* p
 
 	ret->channelMode = pf_utils_get_channel_mode(ps->pdata->config, name);
 	ret->openStatus = CHANNEL_OPENSTATE_OPENED;
+	ret->packetReassembly = (ret->channelMode == PF_UTILS_CHANNEL_INTERCEPT);
+
 	return ret;
 }
 
 static void DynamicChannelContext_free(pServerDynamicChannelContext* c)
 {
-	if (c)
-	{
-		free(c->channel_name);
-		free(c);
-	}
+	if (!c)
+		return;
+
+	if (c->backTracker.currentPacket)
+		Stream_Free(c->backTracker.currentPacket, TRUE);
+
+	if (c->frontTracker.currentPacket)
+		Stream_Free(c->frontTracker.currentPacket, TRUE);
+
+	if (c->channelDataDtor)
```

Comment:
`IFCALL` ?

Reply:
with  ` IFCALL`  arg1 is the callback to test and to run ? Here we're testing `c->frontTracker.currentPacket` and run `Stream_Free`